artifactory/services/download.go

```diff
@@ -220,7 +220,7 @@ func (ds *DownloadService) produceTasks(reader *content.ContentReader, downloadP
 		if err != nil {
 			return "", err
 		}
-		target, placeholdersUsed, err := clientutils.BuildTargetPath(downloadParams.GetPattern(), resultItem.GetItemRelativePath(), downloadParams.GetTarget(), true)
+		target, placeholdersUsed, err := clientutils.BuildTargetPath(downloadParams.GetPattern(), resultItem.GetItemRelativePath(), downloadParams.GetTarget(), true, downloadParams.Regexp)
```

Comment:
Since regular expressions aren't supported for downloads, but only for uploads, it looks like the last argument sent - `downloadParams.Regexp` should be removed.
Notice that the `BuildTargetPath` function isn't used for uploads at all.